### PR TITLE
WIP: Feature: Snapcraft

### PR DESCRIPTION
**Pull request by artemii-yanushevskyi**

### CODE REVIEW

File: content/getting-started/building-a-snap-package.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    environment:
+        SNAPCRAFT_BUILD_ENVIRONMENT: host
+    scripts:
+      - name: Create a snap
+        script: snapcraft snap --output flutter-codemagic-example.snap
+    artifacts:
+        - '**/*.snap'
+```
+
+Additionally, you may want to install the generated `.snap` package onto your machine. The package will not be code signed unless you publish it to Snapcraft. You would need to use the `--dangerous` flag to install the package without code signing:
```
Comment (craastad) [inline content/getting-started/building-a-snap-package.md:39]: Is this for testing locally? If so can you add that here so people know why they would want to do this.

File: content/getting-started/building-a-snap-package.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    snap install your-package.snap --dangerous
+
+In case you are packaging a **Flutter application**, be sure to set `SNAPCRAFT_BUILD_ENVIRONMENT` environment variable to `host`. It is required to avoid virtualization. Read more about virtualization options [here](https://flutter.dev/docs/deployment/linux). Additionally, Snapcraft manages all the dependencies according to `snapcraft.yaml` configuration. There is no need to include Flutter version in `codemagic.yaml`.
+
+{{<notebox>}}
+**Note**: Snap is only available on Linux instances. Make sure to have `instance_type: linux` or `instance_type: linux_x2` in your `codemagic.yaml`.
```
Comment (craastad) [inline content/getting-started/building-a-snap-package.md:46]: You could also link to the linux build machine specs here
https://docs.codemagic.io/specs/versions-linux/

**Review by craastad — COMMENTED** [review]
Some comments to think about.

Question: are the workflow editor docs included in Helina's PR https://github.com/codemagic-ci-cd/codemagic-docs/pull/321?

File: content/getting-started/building-a-snap-package.md (hunk 2 above)
Comment (helinanever) [reply content/getting-started/building-a-snap-package.md]: Done

**Review by helinanever — APPROVED** [review]
LGTM

**Review by craastad — APPROVED** [review]
LGTM :+1:
Feel free to square and merge when it's released.

**Pull request merged**